Project structure diagrams for bulk RNA-seq module

cbethell opened the pull request:
This PR closes #360.
As stated on that issue, this PR is also related to #239.

I took the following steps before filing this PR:
- Updated my local folder structure to reflect what's in `training-modules` and synced the data from S3
- Used the `tree` command (with some tweaks to its options) to get the structure of the `RNA-seq` module (focusing on just the `QC` and `data` directories since the notebooks are explicitly mentioned throughout the module and in the `README.md` and since the files in these directories are the ones that participants will most likely frequent)
- Used the `tree` command to get the structure of the `gastric_cancer` directory within `data`, took a screenshot, and used the supplemental notebook from our 2020-june-training linked in [this comment on #360](https://github.com/AlexsLemonade/training-modules/issues/360#issue-813604302) to annotate this screenshot
- I also updated the folders/files to replace `tximport` with `txi` in preparation for #365, but was not sure if this update should be propagated to the comments on the `gastric_cancer_structure.png` annotated figure since my understanding is that `tximeta` does indeed implement `tximport`. Perhaps you may have some thoughts here @jashapiro.

Questions for reviewers:

- I have placed these figures in the module `README.md` for now. Do we believe this location is best suited for these figures? If not, any suggestions on where they should be placed?
- How do the figures look? Are they clear? Easy to understand? Do they accurately represent the current project structure?

jashapiro commented on RNA-seq/README.md:
```diff
@@ -15,3 +15,11 @@ The notebooks that comprise this module are (estimated time to complete notebook
+
+Beyond the notebooks mentioned above, this module has the following structure:
+
+![module project structure](./diagrams/bulk_project_structure.png)
```
This figure isn't quite up to date, as there will be some changes when #376 goes in, notably to the leukemia directory and the addition of a medulloblastoma directory. Those might be best left collapsed though, as they might add confusion.

jaclyn-taroni commented on RNA-seq/README.md:
```diff
@@ -15,3 +15,11 @@ The notebooks that comprise this module are (estimated time to complete notebook
+
+With the format of each subdirectory in `data` being similar to:
+
+![gastric cancer structure](./diagrams/gastric_cancer_structure.png)
```
Just wanted to note that this still contains a reference to `tximport` which we should fix.

jaclyn-taroni replied:
Ah, I see now - apologies!

> but was not sure if this update should be propagated to the comments on the gastric_cancer_structure.png annotated figure since my understanding is that tximeta does indeed implement tximport

cbethell replied:
No worries! Good to note your stance that it should indeed be updated 👍

cbethell commented:
> This figure isn't quite up to date, as there will be some changes when #376 goes in, notably to the leukemia directory and the addition of a medulloblastoma directory. Those might be best left collapsed though, as they might add confusion.

I've updated this figure now that #376 has been merged and updated the `gastric_cancer_structure.png` to include `tximeta` rather than `tximport`. 

That said, this is ready for a review for which I am going to request you @jashapiro to ensure that I've addressed your comment as expected.

jashapiro reviewed (commented):
A couple more changes: 

In the `data/leukemia` folder, `SRP049821_SraRunTable.txt` should be removed; it was there temporarily, but it should be removed if it is getting there from links. The zebrafish folder should also have a `salmon_quant` directory, but I think it is probably best to just leave that whole folder collapsed; most people won't ever use it as it is only for a supplemental notebook.

The `scripts` directory should also be included, as people will be using that.

Beyond that, we may want to include the `results` and `plots` folders; these may not exist to start, but I think it is probably worth showing people where they will be when we get to them. If we add them, we should include a note that "Some of these directories may be created during the exercises."

cbethell commented:
> A couple more changes:
> 
> In the `data/leukemia` folder, `SRP049821_SraRunTable.txt` should be removed; it was there temporarily, but it should be removed if it is getting there from links. The zebrafish folder should also have a `salmon_quant` directory, but I think it is probably best to just leave that whole folder collapsed; most people won't ever use it as it is only for a supplemental notebook.
> 
> The `scripts` directory should also be included, as people will be using that.
> 
> Beyond that, we may want to include the `results` and `plots` folders; these may not exist to start, but I think it is probably worth showing people where they will be when we get to them. If we add them, we should include a note that "Some of these directories may be created during the exercises."

@jashapiro I collapsed the zebrafish folder, along with the others to keep things _simple_. Let me know what you think about this decision (it does more closely mimic the project structure diagram we had in the supplemental from june training now).

The `index` folder snuck in there as I included the `results` and `plots` folders but I believe this won't hurt (feel free to disagree)! I also added a note re the folders that may not exist to start as you suggested in the `README.md`.

jashapiro commented:
> The `index` folder snuck in there as I included the `results` and `plots` folders but I believe this won't hurt

 I think that is a good thing to have as well!

jashapiro reviewed (approved):
LGTM!

(which isn't to say we might not update it again in the future!)
